src/App.js

```diff
@@ -57,38 +70,58 @@ class App extends React.Component {
 
 
   addTodo = (todoName) => {
+    const { todoItems } = this.state;
     const newTodo = {
       task: todoName,
       id: new Date(),
       completed: false
     };
     this.setState({
-      todoItems: [...this.state.todoItems, newTodo]
+      todoItems: [...todoItems, newTodo]
     });
   };
 
   removeCompleted = () => {
+    const { todoItems } = this.state;
     const newTodoItems = (
-      this.state.todoItems.filter(todo =>
+      todoItems.filter(todo =>
         (todo.completed === false)
       )
     )
     this.setState({todoItems: newTodoItems})
   }
 
   render() {
+    // bonus destructuring state EVERYWHERE:
+    const { searchTerm } = this.state;
+
     return (
       <div className="App">
-        <h1>Not Quite Todoist</h1>
-        <h2>Welcome to your new fav Todo App!</h2>
+        <Jumbotron fluid>
+          <Container fluid>
+          <h1>Not Quite Todoist</h1>
+          <p>Welcome to your new fav Todo App!</p>
+          </Container>
+        </Jumbotron>
+
+        <SearchField
+          type="text"
+          value={searchTerm}
+          onChange={this.editSearchTerm}
+          placeholder="search here"
+        />
+        {/* <input type="text" value={searchTerm} onChange={(e) => this.editSearchTerm(e.target.value)} placeholder="search here" /> */}
         <TodoList
-          todoItems={this.state.todoItems}
+          // todoItems={this.state.todoItems}//now destructured. SOOOO much easier to read & comprehend.
+          todoItems={this.dynamicSearch()}
```

Comment:
Rather than passing in this.state.todoItems & then running the dynamicSearch inside to then render the bits, all TodoList needs to know is the new/temp list. Likewise, dynamicSearch no longer is given the e.target.value to pass back and forth, just the value itself (see declaration above & TodoList.js)